services/workflows-service/src/business-report/business-report.service.ts

```diff
@@ -43,4 +54,87 @@ export class BusinessReportService {
   ) {
     return await this.businessReportRepository.findFirst(args, projectIds);
   }
+
+  async findManyWithFilters(
+    getTransactionsParameters: GetBusinessReportDto,
+    projectId: string,
+    options?: Prisma.BusinessReportFindManyArgs,
+  ) {
+    const args: Prisma.BusinessReportFindManyArgs = {};
+
+    if (getTransactionsParameters.page?.number && getTransactionsParameters.page?.size) {
+      // Temporary fix for pagination (class transformer issue)
+      const size = parseInt(getTransactionsParameters.page.size as unknown as string, 10);
+      const number = parseInt(getTransactionsParameters.page.number as unknown as string, 10);
+
+      args.take = size;
+      args.skip = size * (number - 1);
+    }
+
+    if (getTransactionsParameters.orderBy) {
+      args.orderBy = toPrismaOrderByGeneric(getTransactionsParameters.orderBy);
+    }
+
+    return await this.businessReportRepository.findMany(
+      {
+        ...options,
+        where: {
+          businessId: getTransactionsParameters.businessId,
+        },
+        ...args,
+      },
+      [projectId],
+    );
+  }
+
+  // private buildFilters(filterParams: GetBusinessReportDto): Prisma.TransactionRecordWhereInput {
+  //   const whereClause: Prisma.BusinessReportWhereInput = {};
+  //
+  //   if (filterParams.businessId) {
+  //     whereClause.businessId = filterParams.businessId;
+  //   }
+  //
+  //   if (filterParams.startDate) {
+  //     whereClause.createdAt = {
+  //       ...(whereClause.createdAt as DateTimeFilter),
+  //       gte: filterParams.startDate,
+  //     };
+  //   }
+  //
+  //   if (filterParams.endDate) {
+  //     whereClause.createdAt = {
+  //       ...(whereClause.createdAt as DateTimeFilter),
+  //       lte: filterParams.endDate,
+  //     };
+  //   }
+  //
+  //   if (filterParams.timeValue && filterParams.timeUnit) {
+  //     const now = new Date(); // UTC time by default
+  //     let subtractValue = 0;
+  //
+  //     switch (filterParams.timeUnit) {
+  //       case TIME_UNITS.minutes:
+  //         subtractValue = filterParams.timeValue * 60 * 1000;
+  //         break;
+  //       case TIME_UNITS.hours:
+  //         subtractValue = filterParams.timeValue * 60 * 60 * 1000;
+  //         break;
+  //       case TIME_UNITS.days:
+  //         subtractValue = filterParams.timeValue * 24 * 60 * 60 * 1000;
+  //         break;
+  //       case TIME_UNITS.months:
+  //         now.setMonth(now.getMonth() - filterParams.timeValue);
+  //         break;
+  //       case TIME_UNITS.years:
+  //         now.setFullYear(now.getFullYear() - filterParams.timeValue);
+  //         break;
+  //     }
+  //
+  //     const pastDate = new Date(now.getTime() - subtractValue);
+  //
+  //     whereClause.createdAt = { gte: pastDate };
+  //   }
+  //
```

Comment:
this code is relevant to some table functionality and api filters why is it in comment?

Reply:
its relevant for alerts implementation, can you just remove it from here